vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/PollOptionItem.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.detail.timeline.item
+
+import android.content.Context
+import android.os.Build
+import android.util.AttributeSet
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.view.isVisible
+import im.vector.app.R
+import im.vector.app.core.extensions.setAttributeTintedImageResource
+import im.vector.app.databinding.ItemPollOptionBinding
+
+class PollOptionItem @JvmOverloads constructor(
+        context: Context,
+        attrs: AttributeSet? = null,
+        defStyleAttr: Int = 0
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    interface Callback {
+        fun onOptionClicked()
+    }
+
+    private lateinit var views: ItemPollOptionBinding
+    private var callback: Callback? = null
+
+    init {
+        setupViews()
+    }
+
+    private fun setupViews() {
+        inflate(context, R.layout.item_poll_option, this)
+        views = ItemPollOptionBinding.bind(this)
+
+        views.root.setOnClickListener { callback?.onOptionClicked() }
+    }
+
+    fun update(optionName: String,
```

Comment:
We use `render` for such method, also maybe wrap all those parameter into a data class, or a sealed class with all the possible case would be easier to understand and maintain.
Setting the callback in the same method is also a bit weird, but ok

Reply:
Refactored, i think it is cleaner now.